components/scoped-notification/index.jsx

```diff
@@ -21,24 +24,31 @@ const propTypes = {
 		PropTypes.string,
 	]),
 	/**
-	 *  Theme for the scoped notification
+	 *  The icon to be displayed in the scoped notification. Accepts an `Icon` component
 	 */
-	theme: PropTypes.oneOf(['dark', 'light']),
+	icon: PropTypes.node,
 	/**
-	 *  Icon for the scoped notification. This is currently limited to the utility set of icons.
+	 *  Theme for the scoped notification
 	 */
-	iconName: PropTypes.string,
+	theme: PropTypes.oneOf(['dark', 'light']),
 };
 
-const defaultProps = {
-	iconName: 'info',
-};
+const defaultProps = {};
 
 /**
  * A Scoped Notification Component serve advisory information for the user that is not important enough to justify an alert.
  */
 class ScopedNotification extends React.Component {
+	constructor(props) {
+		super(props);
+		checkProps(SCOPED_NOTIFICATION, props, componentDoc);
+	}
+
 	render() {
+		// For backwards compatibility
+		const iconAssistiveText =
+			(this.props.assistiveText && this.props.assistiveText.icon) || 'Info';
```

Comment:
I'm not seeing `this.props.assistiveText.icon` in the propTypes and am confused.

I'm thinking if the prop is undefined, nothing should be present. In my original comment, I was saying that the snapshot wasn't passing. I wasn't saying that the previous snapshot was correct.

This seems like it should be a defaultValue prop and not defined in the render function. What's the reason for it not being a default prop?

Reply:
@interactivellama I was struggling with what to do with this... in the original code, `this.props.assistiveText.icon` was incorrectly passed to the `Icon` prop as-is (`Icon` only accepts a `label` assistiveText option) but not listed in PropTypes or in defaultProps. Now that we have an `icon` prop, assistiveText should be passed in to the `Icon` component itself, which would make `this.props.assistiveText.icon` redundant... but we still have an automatically generated info Icon with assistiveText even if the `icon` prop is not set so I suppose that should still be exposed 😕

I'm thinking the following should work:
- Expose `assistiveText.icon` in the `PropTypes` so that people are aware it's there but don't set a defaultValue
- For cases where `props.icon` is present, clone the `Icon` component and it's props with an assistiveText override that merges the Icon assistiveText object on top of a compatible ScopedNotification assistiveText object
- For cases where `props.icon` is not present, assistiveText will be determined by `(this.props.assistiveText && this.props.assistiveText.icon) || 'Info'` as before

I'll push an update with that approach and we can decide whether to move forward with it or iterate again then.